xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -97,11 +97,19 @@ void CGUIWindowPVRGuide::StopRefreshTimelineItemsThread()
 
 void CGUIWindowPVRGuide::RegisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
   g_EpgContainer.RegisterObserver(this);
+  if (g_PVRTimers)
+    g_PVRTimers->RegisterObserver(this);
+  CGUIWindowPVRBase::RegisterObservers();
 }
 
 void CGUIWindowPVRGuide::UnregisterObservers(void)
 {
+  CSingleLock lock(m_critSection);
+  CGUIWindowPVRBase::UnregisterObservers();
+  if (g_PVRTimers)
```

Comment:
what do you think about this approach, which does not need window messages:

pvr components will be no longer Observables, instead of calling NotifyObservers() directly like they do now, they will call g_PVRManager.NotifyObservers(), passing the same observable message like they do today. the pvr windows no longer observe different pvr components, but the pvr manager. problem solved.